crates/sequencer/src/sequencer.rs

```diff
@@ -922,10 +922,15 @@ where
         let mut missed_da_blocks_count = 0;
 
         loop {
-            let mut interval = tokio::time::interval(target_block_time - parent_block_exec_time);
-            // The first ticket completes immediately.
-            // See: https://docs.rs/tokio/latest/tokio/time/struct.Interval.html#method.tick
-            interval.tick().await;
+            let block_production_tick = tokio::time::sleep(
+                target_block_time
+                    .checked_sub(
+                        parent_block_exec_time
+                            .checked_sub(target_block_time)
+                            .unwrap_or_default(),
+                    )
+                    .unwrap_or_default(),
```

Comment:
When parent_block_exec_time < target_block_time, we will always wait for 2 seconds (target_block_time), is that expected behavior?

Reply:
`target_block_time - (parent_block_exec_time - target_block_time)` ?

Reply:
> When parent_block_exec_time < target_block_time, we will always wait for 2 seconds (target_block_time), is that expected behavior?

yes